lib/DDGC/Web/Controller/InstantAnswer.pm

```diff
@@ -701,6 +703,44 @@ sub commit_save :Chained('commit_base') :PathPart('save') :Args(0) {
     return $c->forward($c->view('JSON'));
 }
 
+# Install the data on the beta server
+# Data is an hash of arrays
+# {goodies: [#pr1, #pr2, #pr3], spice: [#pr1, #pr2] ... and so on
+sub send_to_beta :Chained('base') :PathPart('send_to_beta') :Args(0) {
+    my ( $self, $c ) = @_;
+    
+    my $ua = LWP::UserAgent->new;
+
+    my $result = '';
+    $c->stash->{x}->{result} = $result;
+    return $c->forward($c->view('JSON')) unless ($c->req->params->{data} && $c->user && $c->user->admin);
+
+    my $server = "http://beta.duckduckgo.com/install";
+    my $key = $ENV{'BETA_KEY'};
+    warn $key;
+
+    for my $data (@{$c->req->params->{data}}) {
+        my $req = HTTP::Request->new(GET => $server);
+        my $header_data = "sha1=".Digest::SHA::hmac_sha1_hex($data, $key);
```

Comment:
It's annoying but you will need to decode the json string to iterate above and then encode it again for the digest.  `hmac_sha1_hex( to_json($data, $key))` or something like that